digest/src/digest.rs

```diff
@@ -196,21 +196,15 @@ impl<D: Update + FixedOutputReset + Reset + Clone + 'static> DynDigest for D {
     }
 
     fn finalize_into(self, buf: &mut [u8]) -> Result<(), InvalidBufferSize> {
-        if buf.len() == self.output_size() {
-            FixedOutput::finalize_into(self, Output::<Self>::from_mut_slice(buf));
-            Ok(())
-        } else {
-            Err(InvalidBufferSize)
-        }
+        let buf = <&mut Output<Self>>::try_from(buf).map_err(|_| InvalidBufferSize)?;
+        FixedOutput::finalize_into(self, buf);
+        Ok(())
```

Comment:
I think it's better to write it like this:
```rust
let buf: &mut Output<Self> = buf.try_into().map_err(|_| InvalidBufferSize)?;
```
I think you can even skip the type ascription, since compiler should be able to determine the type itself.

Another alternative:
```rust
buf
    .try_into()
    .map_err(|_| InvalidBufferSize)
    .map(|buf| FixedOutput::finalize_into(self, buf))
```

Reply:
Applied the latter suggestion in 6a2190a